src/Arcus.Security.Core/Storage/Source/SecretStoreBuilderExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using GuardNet;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+
+namespace Arcus.Security.Core.Storage.Source
+{
+    /// <summary>
+    /// Extends the <see cref="SecretStoreBuilder"/> to provide additional secret sources.
+    /// </summary>
+    public static class SecretStoreBuilderExtensions
+    {
+        /// <summary>
+        /// Adds a secret source to the secret store of the application that gets its secrets from the environment.
+        /// </summary>
+        /// <param name="builder">The builder to create the secret store.</param>
+        /// <param name="target">The target on which the environment variables should be retrieved.</param>
+        /// <param name="version">The fixed version to attach to each environment variable.</param>
+        public static SecretStoreBuilder AddEnvironmentVariables(
+            this SecretStoreBuilder builder,
+            EnvironmentVariableTarget target = EnvironmentVariableSecretProvider.DefaultTarget,
+            string version = EnvironmentVariableSecretProvider.DefaultVersion)
```

Comment:
What is this? Is this for Secret.Version or? Because this seems confusing and would change Secret to have optional secret instead

Reply:
Yes, the `Secret.Version`. It seemed "wrong" to hardcode the version in both the environment and configuration secret providers, so I added a default one and created the opportiunity to override it.

Reply:
I think it's best to make `Version` optional for `Secret` as not every provider has support for that

Reply:
Yes, ok.